### PR TITLE
Docs/contributors  adding projects

### PR DESCRIPTION
**Pull request by cereallarceny**
Attempting to figure out how we may better guide users from "I want to help on this project" to actually joining a specific team or part of the OpenMined ecosystem.  How do we direct people where they need to go to start helping?... rather than assume they'll figure it out on their own by reading Slack.

[See Issue 36.](https://github.com/OpenMined/Docs/issues/36)

### CODE REVIEW

**Comment by cereallarceny** [comment]
I've gone ahead and filled out "Machine Learning" as a potential example... we'll want to document the others.

File: contributing/quickstart.md [hunk 1]
```diff
@@ -18,29 +18,94 @@ This video covers the project as a whole including high level goals and the broa
 
 - [OpenMined Introduction Slides](https://docs.google.com/presentation/d/1Tm7jh3pChks0ObJNb7x2ZZhhVOJhmCVN2KDPWyZohQU/edit)
 
-## Step 3: Syft Installation + Homomorphic Encryption Demo
+## Step 3: Figure out where you might want to contribute
+
```
Comment (anoff) [inline contributing/quickstart.md]: Step 3 is there twice. I assume the first one is legacy? :)

File: contributing/quickstart.md [hunk 2]
```diff
@@ -18,29 +18,94 @@ This video covers the project as a whole including high level goals and the broa
...
+**Slack channels:** #sonar, #syft
+
+#### Homomorphic Encryption
+FILL ME OUT
+
+#### Blockchain
+FILL ME OUT
+
+#### Node.js Development
```
Comment (anoff) [inline contributing/quickstart.md]: Programming language and domain seem to be non-exclusive. Do you think putting the whole thing in a table would be possible?

for example:

| # | python | nodeJS | react |
|:----------------------:|----------------|--------------------|----------|
| blockchain | pySonar | mine.js capsule.js | - |
| homomorphic encryption | pySyft | capsule.js | - |
| UX | sonar notebook | adapters | sonar UI |

File: contributing/quickstart.md [hunk 3]
```diff
@@ -18,29 +18,94 @@ This video covers the project as a whole including high level goals and the broa
...
+FILL ME OUT
+
+#### Python Development
+**Repositories:**
+- [PySonar](https://github.com/OpenMined/PySonar) - Python client to blockchain application (Sonar)
+- [PySyft](https://github.com/OpenMined/PySyft) - Python deep learning library
+- [Capsule](https://github.com/OpenMined/Capsule) - Blockchain oracle for time released public/private key generation/interaction
```
Comment (anoff) [inline contributing/quickstart.md]: it's gonna be `capsule.js` 👯

**Comment by kevinahuber** [comment]
I would leave this out of the quickstart, and make it a link outside of it. I think its SUPER overwhelming to have the project as a whole, and the quickstart is like the tutorial level to OpenMined. You kill some skeletons, make a fire, do your first spell, _then_ receive a map from uncle Patrick upon completion.

**Comment by cereallarceny** [comment]
Fair enough, that's not a bad idea.  Maybe treat it as a "token of accomplishment"?  The only problem I see with that is it requires people to have a knowledge of Python and machine learning.  Most front-end developers, much less designers, UX specialists, lawyers, and marketers will be able to get past that though.  I would be concerned that could be a massive bottleneck for less technical individuals.

**Comment by cereallarceny** [comment]
@iamtrask I added the other sections in, but I'm not sure if I got all the right repos in the right place.  Can I get a quick copy check to make sure all the information is factual?  Also note, we don't have a sonar.js library so that's missing a link until further notice.

**Comment by cereallarceny** [comment]
@swaroopch I implemented the changes to the Slack channel names as discussed.  Would you mind confirming this in my latest commit?

File: contributing/quickstart.md [hunk 4]
```diff
@@ -18,29 +18,106 @@ This video covers the project as a whole including high level goals and the broa
...
-## Step 3: Syft Installation + Homomorphic Encryption Demo
+## Step 3: Figure out where you want to contribute!
+
+We welcome all people willing to contribute of any skill set and level of experience. The following is a list of consistently needed skillsets:
+
+### Machine learning
+**Repositories:**
+- [PySonar](https://github.com/OpenMined/PySonar) - Python client to blockchain application (Sonar)
+- [sonar.js](NEEDS A LINK) - Node.js client to blockchain application (Sonar)
+- [Sonar](https://github.com/OpenMined/Sonar) - Python Eterheum blockchain implementation
```
Comment (swaroopch) [inline contributing/quickstart.md]: typo - `s/Eterheum/Ethereum`

File: contributing/quickstart.md [hunk 5]
```diff
@@ -18,29 +18,106 @@ This video covers the project as a whole including high level goals and the broa
...
+### Homomorphic Encryption
+**Repositories:**
+- [PySyft](https://github.com/OpenMined/PySyft) - Python deep learning library
+- [syft.js](https://github.com/OpenMined/syft.js) - Node.js deep learning library (for web and mobile browsers)
+
+**Slack channels:** #team_syft
+
+### Blockchain
+**Repositories:**
+- [Sonar](https://github.com/OpenMined/Sonar) - Python Eterheum blockchain implementation
```
Comment (swaroopch) [inline contributing/quickstart.md]: typo - s/Eterheum/Ethereum

File: contributing/quickstart.md [hunk 6]
```diff
@@ -18,29 +18,106 @@ This video covers the project as a whole including high level goals and the broa
...
+- [PySonar](https://github.com/OpenMined/PySonar) - Python client to blockchain application (Sonar)
+- [sonar.js](NEEDS A LINK) - Node.js client to blockchain application (Sonar)
```
Comment (swaroopch) [inline contributing/quickstart.md]: Remove the link for now?

File: contributing/quickstart.md [hunk 7]
```diff
@@ -18,29 +18,106 @@ This video covers the project as a whole including high level goals and the broa
...
+
+### DevOps
+**Repositories:** None
+
+**Slack channels:** #team_devops
+
+### Technical Writing
+**Repositories:** None
+
+**Slack channels:** #terminology, #process, #team_docs
```
Comment (swaroopch) [inline contributing/quickstart.md]: Remove `#terminology` - no longer exists

**Review by swaroopch — COMMENTED** [review]
LGTM!

File: contributing/quickstart.md [hunk 8]
```diff
@@ -18,29 +18,106 @@ This video covers the project as a whole including high level goals and the broa
...
+
+**Slack channels:** #team_sonar, #team_syft, #team_capsule
+
+### React.js Development
+**Repositories:** Coming soon.
+
+**Slack channels:** #team_ui
+
+### DevOps
+**Repositories:** None
```
Comment (swaroopch) [inline contributing/quickstart.md]: say "all" instead?

**Comment by swaroopch** [comment]
@cereallarceny Do we want to mention `#beginner` and `#questions` channels?

**Comment by cereallarceny** [comment]
Good idea, definitely.  Working on a commit for all these changes.

**Comment by cereallarceny** [comment]
Approve the review and merge me in boss!

File: contributing/quickstart.md [hunk 9]
```diff
@@ -25,8 +25,8 @@ We welcome all people willing to contribute of any skill set and level of experi
 ### Machine learning
 **Repositories:**
 - [PySonar](https://github.com/OpenMined/PySonar) - Python client to blockchain application (Sonar)
-- [sonar.js](NEEDS A LINK) - Node.js client to blockchain application (Sonar)
-- [Sonar](https://github.com/OpenMined/Sonar) - Python Eterheum blockchain implementation
+- [sonar.js (Link pending) - Node.js client to blockchain application (Sonar)
```
Comment (swaroopch) [inline contributing/quickstart.md]: extra leading square bracket??

File: contributing/quickstart.md [hunk 10]
```diff
@@ -41,17 +41,17 @@ We welcome all people willing to contribute of any skill set and level of experi
 
 ### Blockchain
 **Repositories:**
-- [Sonar](https://github.com/OpenMined/Sonar) - Python Eterheum blockchain implementation
+- [Sonar](https://github.com/OpenMined/Sonar) - Python Ethereum blockchain implementation
 - [PySonar](https://github.com/OpenMined/PySonar) - Python client to blockchain application (Sonar)
-- [sonar.js](NEEDS A LINK) - Node.js client to blockchain application (Sonar)
+- [sonar.js (Link pending) - Node.js client to blockchain application (Sonar)
```
Comment (swaroopch) [inline contributing/quickstart.md]: extra leading square bracket??

File: contributing/quickstart.md [hunk 11]
```diff
@@ -93,7 +93,7 @@ There's a plethora of legal issues that come with a person taking back ownership
 We have a highly capable set of creative people working on the brand of OpenMined. How do we get this brand to be viewed positively by developers? How would a rollout to the general public work? How much time should an open source project spend on marketing itself?
 
 ### Anything else?
-We'd love to have you! The only requirement for joining the OpenMined contributor network is a desire to follow the core values and vision (NEEDS A LINK) of the project. Feel free to reach out in the general channel on Slack and someone will be sure to point you in the right direction!
+We'd love to have you! The only requirement for joining the OpenMined contributor network is a desire to follow the core values and vision of the project. Feel free to reach out in the #general, #beginner, and #questions channel on Slack and someone will be sure to point you in the right direction!
```
Comment (swaroopch) [inline contributing/quickstart.md]: I believe it is called `#general-discussion` and not `#general`?

**Comment by swaroopch** [comment]
> Approve the review and merge me in boss!

I can do that!? 😮

Just one blocking change - about `#general-discussion` (not `#general`) and we should be good to merge 🙂

**Comment by cereallarceny** [comment]
Pushed up those changes, sir!  I think you can review it if you want.  We can't merge until a review has been approved.  :)

**Review by swaroopch — APPROVED** [review]

**Comment by swaroopch** [comment]
@cereallarceny github says "At least one approved review is required by reviewers with write access." - I guess I don't have write access :)

**Comment by cereallarceny** [comment]
Dang it!  @iamtrask We need your help!

**Review by anoff — APPROVED** [review]
me to the rescue \o/

**Review by iamtrask — CHANGES_REQUESTED** [review]
I second the notion that this is very intimidating as a document. Step 3 should instead be the last step, and it should broken out into sub pages (sortof like the old Goosebumps books!...choose your own ending).

**Comment by cereallarceny** [comment]
Why do we want to put these at the end @iamtrask?  What about for people who aren't technical or aren't Python developers?  I can't imagine them being very interested in spinning up Sonar if they only have UI design expertise.  I think that narrows the scope of interested individuals significantly -
 maybe we should appeal to less technically savvy individuals as well?

**Review by kevinahuber — CHANGES_REQUESTED** [review]
@cereallarceny brings up a good point, but I don't think the answer is to put this part in the middle. 
This quickstart is best for particularly python developers and the first components we had code for. After Hydrogen, the quickstart  _should_ be as simple as installing docker and running `docker-compose up`. BUT, we are not there yet and I think this is a good add to merge soon.

The solution I would like best (and what I think meets most sentiments):
- [ ] Move Step 3 to the final Step
- [ ] Include a note at the end of Step 2 that reads(ish):

> Great! You've made it this far. Steps 3-6 get pretty weedsy into code. If you are not a developer and would still love to contribute, hop down to [Step 7](#whooop-this-isnt-a-real-link)

Then in the future Steps 3-6 will be increasingly non-dev friendly. Thoughts?

@iamtrask What are your goals for this document?

**Comment by cereallarceny** [comment]
I think that's a great idea @kevinahuber - seems like a good balance.  I'll implement this now.

**Review by kevinahuber — APPROVED** [review]
👍

**Review by iamtrask — APPROVED** [review]
beautiful!

**Pull request merged**